books/nlp/src/llms/agents/README.md

```diff
@@ -1 +1,119 @@
+and even [other agents](./multi_agents.md).
+They should fulfill a clear objective which is clearly communicated
+to the LLM through a short, formatted description.
+This is so the LLM knows of the tool's existence and can invoke it when
+necessary. However, the input and output of an LLM is text, so what invoking
+a tool actually entails is specifying that the LLM should output their
+invoking of a tool in a set format,
+with common formats including JSON or direct code.
+
+Memory: The information retrieved from a tool updates the agent's memory,
+which also contains the context of its objective: the original
+user-defined objective, any other user input, as well as the results
+of prior planning, actions, and reflection.
+This memory is vital for the agent to complete its objective coherently
+and not be stuck in endless loops, conducting unnecessary actions,
+or offering irrelevant results.
+
+LLM: The underlying LLM powering the agent's every move can be
+any language model, including the notable models listed in
+[Section 2](../../models/README.md).
+
+Framework: The final component of an agent that cannot be overlooked is the
+"connective tissue" that enables the LLM to work together with its memory
+and its tools. This is the code that structures the user-defined objective
+and the tool descriptions into a prompt for the LLM, that parses an LLM's
+output and directs their JSON/code to the corresponding tool, that incorporates
+tool results into memory and text to properly give back to
+the LLM for reflection.
+
+## Applications
+
+The potential applications for agents are quite broad. Some examples include:
+
+- Personal scheduling assistant
+- Customer service queue specialist
+- Internet-of-things (IoT) hub manager
+- Discussion forum moderator
+- Lab research assistant
+- etc.
+
+While the above are theoretical applications, tangible agents have also begun
+making their way to the market. At the time of writing this reference,
+Deep Research (2025), a research assistant for synthesizing literature,
+and Manus (2025), for broader analysis and development,
+are two agents that have made headlines. Furthermore, Casper et al. (2025)
+have compiled an index to track existing agents and discover patterns. They
+have found agents "being deployed at a steadily increasing rate".
+
+## Limitations
+
+Built on LLMs, agents can be computationally expensive,
+and thus may be unnecessary for overly simplistic tasks.
+Due to their semi-supervised nature, the possibility of an agent
+making inefficient LLM calls or worse, getting stuck in feedback loops,
+should be kept in mind.
+Extra care should be taken when implementing multi-agent collaboration,
+as a hallucination in one agent would affect the whole system.
+Additionally with multi-agents, if they were all built on the same LLM,
+any reasoning deficiency would be shared across all agents.
+Lastly, more work and time is needed to foster trust in the community
```

Comment:
Another readability edit: "Lastly, more work and time are needed to foster community trust in the viability of agents for everyday life."